internal/compiler-bridge/src-2.10/main/scala/xsbt/Dependency.scala

```diff
@@ -156,6 +156,12 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile {
       }
     }
 
+    private def addTreeDependency(tree: Tree): Unit = {
+      addDependency(tree.symbol)
+      if (tree.tpe != null)
+        symbolsInType(tree.tpe).foreach(addDependency)
```

Comment:
This probably makes the Dependency phase significantly slower. I think we'll need to remember what types we visited already. I'm ok merging this because switch to class-based dependency tracking made the performance worse and we'll need to do some profiling.